### PR TITLE
[SDK-1620]-Remove PII data using `removeValueForKey`

### PR DESCRIPTION
**Pull request by william-ct**
Remove Specific User Properties from databases, Shared Preference, and elsewhere(arp etc) stored in the SDK ensuring no impact on Identity Management / onUserLogin.

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java [hunk 1]
```diff
@@ -2163,7 +2163,8 @@ public void removeMultiValuesForKey(final String key, final ArrayList<String> va
     }
 
     /**
-     * Remove the user profile property value specified by key from the user profile
+     * Remove the user profile property value specified by key from the user profile. Alternatively this method
+     * can also be used to remove PII data (for eg. Email,Name,Phone), locally.
```
Comment (william-ct) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java:2167]: check description

Comment (william-ct) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java:2167]: locally from database and shared prefs

File: clevertap-core/src/test/java/com/clevertap/android/sdk/LoginInfoProviderTest.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+        val sharedPreferences = appCtx.getSharedPreferences("WizRocket", Context.MODE_PRIVATE)
+
+
+        assertEquals("{\"email_abc@gmail.com\":\"\"}",
+            sharedPreferences.getString("cachedGUIDsKey:id",""))
+    }
+
+    @Test
+    fun test_removePIICacheGUIDForIdentifier_when_cache_data_contains_email_and_identity_removes_key() {
```
Comment (william-ct) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/LoginInfoProviderTest.kt:102]: remove_provided_key

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

**Comment by william-ct** [comment]
> versioning, release notes, doc changes

Will be adding all of the above changes in a new branch checked out from develop

**Pull request merged**